sonar-dotnet-shared-library/pom.xml

```diff
@@ -20,34 +20,47 @@
     <!-- used for deployment to SonarSource Artifactory -->
     <gitRepositoryName>sonar-dotnet-shared-library</gitRepositoryName>
     <sonar.exclusions>target/generated-sources/protobuf/**</sonar.exclusions>
-    <sonar.analyzer.commons>1.10.2.456</sonar.analyzer.commons>
+    <sonar.analyzer.commons>1.14.1.690</sonar.analyzer.commons>
   </properties>
 
   <dependencies>
+    <!-- provided at runtime -->
     <dependency>
       <groupId>org.sonarsource.sonarqube</groupId>
       <artifactId>sonar-plugin-api</artifactId>
       <version>${sonar.version}</version>
       <scope>provided</scope>
     </dependency>
-
     <dependency>
       <groupId>com.google.code.findbugs</groupId>
       <artifactId>jsr305</artifactId>
       <version>3.0.2</version>
       <scope>provided</scope>
     </dependency>
+
+    <!-- Compile dependencies -->
     <dependency>
       <groupId>com.google.code.gson</groupId>
       <artifactId>gson</artifactId>
       <version>2.7</version>
     </dependency>
-
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
       <version>${protobuf.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.sonarsource.analyzer-commons</groupId>
+      <artifactId>sonar-analyzer-commons</artifactId>
+      <version>${sonar.analyzer.commons}</version>
+      <scope>compile</scope>
```

Comment:
_For consistency:_ The other occurrences of compiled dependencies do not use explicit value (like the one just above this, or the modified ones in other two POMs)
```suggestion
```